### ISSUE TITLE
How do I use the lwc logger with the Nebula namespace?

### ISSUE THREAD

**Issue by joeythomaschaske** [jongpie/NebulaLogger#533]
### New Feature Summary

I'm following the example in the readme and substituting `c` with `Nebula` in my lwc component, so the lwc contains

`<Nebula-logger></Nebula-logger>`

When I deploy this I get the error `No MODULE named Nebula:logger found`

What is the correct way to use this in an lwc?

**Comment by jefersonchaves** [comment]
Hello Joseph,

1. What version of Managed Package have you installed?
2. Are you using inside another managed package or is it a regular org?

**Comment by joeythomaschaske** [comment]
Closing this, I was using this in a scratch org with a namespace which I believe was the problem. 

Though with LWS enabled it [should allow](https://salesforce.stackexchange.com/questions/408083/can-i-use-an-externally-namespaced-component-in-my-namespace) you to do this, but I didn't have any luck. I was doing a POC so it's not entirely pressing but maybe this issue will help someone else down the road

**Comment by jongpie** [comment]
@joeythomaschaske thanks for the follow up info, and apologies for previously overlooking this issue! 

I'm going to re-open this issue, and use it to track progress on documenting this in Nebula Logger's help docs (which I'm working on overhauling). Testing of the LWC in the managed package is definitely an area I've somewhat neglected due to some complexities in setting up a testing org + effort in manually testing some of the functionality, so I'll spend some time working on documenting what is/isn't supported, and examples of how to implement it in your own LWCs.

**Comment by jongpie** [comment]
@joeythomaschaske this is now officially a bug! 🥳 😭  

I've talked with a few other people that have also tried to get the namespaced LWC to work, all without success. I've done a few beta versions of the managed package to explore some options, and the issue seems to be that LWC is not currently fully exposed in the namespaced package.Specifically, in the LWC's config file (`logger.js-meta.xml`), [the `isExposed` node](https://github.com/jongpie/NebulaLogger/blob/v4.11.5/nebula-logger/core/main/logger-engine/lwc/logger/logger.js-meta.xml#L4) needs to be set to `true` for namespaced components so that they can be used from outside the namespace.

With this change in place, I made a beta version of the managed package that I was able to successfully import & reference, using the JavaScript below. Embedding the LWC in markup, using `<Nebula-logger></Nebula-logger>`, also appears to now work.

```javascript
import { LightningElement } from 'lwc';
import { createLogger } from 'Nebula/logger';

export default class ManagedPackageTest extends LightningElement {
    logger = createLogger();

    handleClick() {
        this.logger.info('Hello World 🙋‍♀️');
        this.logger.saveLog();
    }
}

```

I'm still doing some more tests to confirm that there aren't any other issues - but my hope is that [the upcoming release `v4.12.0`](https://github.com/jongpie/NebulaLogger/milestone/12) will resolve this issue.

**Comment by jongpie** [comment]
Another update on this - I am unfortunately going to postpone exposing the `logger` LWC in the managed package for now, it won't be included in the `v4.12.0` release 😞. There are currently 2 open issues for the component (issue #346 has been reopened, and discussion #581 should have been resolved in [`v4.11.12`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.11.12)) that are causing problems with some core logging functionality for JavaScript. Releasing/exposing it now in the managed package would be a... not-ideal experience, so I'm going to wait until those issues have first been fully resolved in the unlocked package. Hopefully, everything will be ready for [the `v4.13.0` Spring '24 milestone](https://github.com/jongpie/NebulaLogger/milestone/13).
